extensions/sql-database-projects/src/test/project.test.ts

```diff
@@ -1327,6 +1327,101 @@ describe('Project: properties', function (): void {
 		should(() => project.getDatabaseDefaultCollation())
 			.throw('Invalid value specified for the property \'DefaultCollation\' in .sqlproj file');
 	});
+
+	it('Should add database source to project property', async function (): Promise<void> {
+		projFilePath = await testUtils.createTestSqlProjFile(baselines.sqlProjectInvalidCollationBaseline);
+		const project = await Project.openProject(projFilePath);
+
+		// Should add a single database source
+		await project.addDatabaseSource('test1');
+		let databaseSourceItems: string[] = project.getDatabaseSourceValues();
+		should(databaseSourceItems.length).equal(1);
+		should(databaseSourceItems[0]).equal('test1');
+
+		// Should not add database source values with semicolon
+		await project.addDatabaseSource(';test2;');
+		databaseSourceItems = project.getDatabaseSourceValues();
+		should(databaseSourceItems.length).equal(1);
+
+		// Should add multiple database sources
+		await project.addDatabaseSource('test2');
+		await project.addDatabaseSource('test3');
+		databaseSourceItems = project.getDatabaseSourceValues();
+		should(databaseSourceItems.length).equal(3);
+		should(databaseSourceItems[0]).equal('test1');
+		should(databaseSourceItems[1]).equal('test2');
+		should(databaseSourceItems[2]).equal('test3');
+
+		// Should not add duplicate database sources
+		await project.addDatabaseSource('test1');
+		await project.addDatabaseSource('test2');
+		await project.addDatabaseSource('test3');
+		should(databaseSourceItems.length).equal(3);
+		should(databaseSourceItems[0]).equal('test1');
+		should(databaseSourceItems[1]).equal('test2');
+		should(databaseSourceItems[2]).equal('test3');
+		should(getWellKnownDatabaseSourceString(project.getDatabaseSourceValues())).equal('');
+
+		await project.addDatabaseSource(constants.WellKnownDatabaseSources[0]);
+		should(getWellKnownDatabaseSourceString(project.getDatabaseSourceValues())).equal(constants.WellKnownDatabaseSources[0]);
```

Comment:
These are more testing the notion of well-known source strings; probably should put them in a separate test case